verify/image_test.go

```diff
@@ -0,0 +1,114 @@
+package verify
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("run image tests", Ordered, func() {
+	var (
+		machineName string
+		session    *machineSession
+	)
+	
+	BeforeAll(func() {
+		machineName, session, err = mb.initNowWithName()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+		DeferCleanup(func() {
+			// Stop machine
+			stopMachineCmd := []string{"machine", "stop", machineName}
+			StopMachineSession, err := mb.setCmd(stopMachineCmd).run()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(StopMachineSession).To(Exit(0))
+	
+			// Remove machine
+			removeMachineCmd := []string{"machine", "rm", "-f", machineName}
+			removeMachineSession, err := mb.setCmd(removeMachineCmd).run()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(removeMachineSession).To(Exit(0))
+		})
+	})
+	
+	Context("when the podman machine is up and running", func() {
+		It("should serve the Podman rootful API", func() {
+			curlCmd := []string{"machine", "ssh", machineName, "sudo", "curl", "-sS", "--unix-socket", "/run/podman/podman.sock", "http://d/_ping"}
+			curlSession, err := mb.setCmd(curlCmd).run()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(curlSession).To(Exit(0))
+			Expect(curlSession.outputToString()).To(Equal("OK"))
+		})
+		It("should serve the Podman rootless API", func() {
+			curlCmd := []string{"machine", "ssh", machineName, "curl", "-sS", "--unix-socket", "/run/user/1000/podman/podman.sock", "http://d/_ping"}

```

Comment:
I don't think the hard coded 100 uid is right, AFAIK podman-machine keeps the uid from the host user.

Reply:
Ok I will update this then. I think I should keep it hardcoded on Windows though.

Reply:
yeah looking at podman we use
```
	uid := os.Getuid()
	if uid == -1 { // windows compensation
		uid = 1000
	}
```

So I guess you can copy that here to the tests

Reply:
Done.